### PR TITLE
TOOLS-2921 Update to Go Driver 1.6

### PR DESCRIPTION
**Pull request by tfogo**
In [GODRIVER-1881](https://jira.mongodb.org/browse/GODRIVER-1881), `EstimatedDocumentCount()` uses `$collStats` which will not work for views. So `DeferredQuery.Count()` now uses `CountDocuments()` for views.

### CODE REVIEW

**Review by huan-Mongo — APPROVED** [review]
LGTM!

**Review by dragonlee8 — APPROVED** [review]
LGTM!

**Review by rychipman — APPROVED** [review]
revendor LGTM, but if we could fix the `modtidy` failure before merging, that would be great!

**Pull request merged**
